src/Nest/Mapping/Types/Core/DateNanos/DateNanosProperty.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.Serialization;
+using Elasticsearch.Net.Utf8Json;
+
+namespace Nest
+{
+	[InterfaceDataContract]
+	public interface IDateNanosProperty : IDocValuesProperty
+	{
+		/// <summary>
+		/// Mapping field-level query time boosting. Accepts a floating point number, defaults to 1.0.
+		/// </summary>
+		[DataMember(Name = "boost")]
+		double? Boost { get; set; }
+
+		/// <summary>
+		/// The date format(s) that can be parsed. Defaults to strict_date_optional_time||epoch_millis.
+		/// <see cref="DateFormat" />
+		/// </summary>
+		[DataMember(Name = "format")]
+		string Format { get; set; }
+
+		/// <summary>
+		/// If true, malformed numbers are ignored. If false (default), malformed numbers throw an exception
+		/// and reject the whole document.
+		/// </summary>
+		[DataMember(Name = "ignore_malformed")]
+		bool? IgnoreMalformed { get; set; }
+
+		/// <summary>
+		/// Should the field be searchable? Accepts true (default) and false.
+		/// </summary>
+		[DataMember(Name = "index")]
+		bool? Index { get; set; }
+
+		/// <summary>
+		/// Accepts a date value in one of the configured format's
+		/// as the field which is substituted for any explicit null values. Defaults to null,
+		/// which means the field is treated as missing.
+		/// </summary>
+		[DataMember(Name = "null_value")]
+		DateTime? NullValue { get; set; }
+	}
+
+	[DebuggerDisplay("{DebugDisplay}")]
+	public class DateNanosProperty : DocValuesPropertyBase, IDateNanosProperty
+	{
+		public DateNanosProperty() : base(FieldType.DateNanos) { }
+
+		/// <inheritdoc />
+		public double? Boost { get; set; }
+
+		/// <inheritdoc />
+		public string Format { get; set; }
+
+		/// <inheritdoc />
+		public bool? IgnoreMalformed { get; set; }
+
+		/// <inheritdoc />
+		public bool? Index { get; set; }
+
+		/// <inheritdoc />
+		public DateTime? NullValue { get; set; }
+
+		/// <inheritdoc />
+		public int? PrecisionStep { get; set; }
+	}
+
+	[DebuggerDisplay("{DebugDisplay}")]
+	public class DateNanosPropertyDescriptor<T>
+		: DocValuesPropertyDescriptorBase<DateNanosPropertyDescriptor<T>, IDateNanosProperty, T>, IDateNanosProperty
+		where T : class
+	{
+		public DateNanosPropertyDescriptor() : base(FieldType.DateNanos) { }
+
+		/// <inheritdoc cref="IDateNanosProperty.Boost"/>
+		double? IDateNanosProperty.Boost { get; set; }
+
+		/// <inheritdoc cref="IDateNanosProperty.Format"/>
+		string IDateNanosProperty.Format { get; set; }
+
+		/// <inheritdoc cref="IDateNanosProperty.IgnoreMalformed"/>
+		bool? IDateNanosProperty.IgnoreMalformed { get; set; }
+
+		/// <inheritdoc cref="IDateNanosProperty.Index"/>
+		bool? IDateNanosProperty.Index { get; set; }
+
+		/// <inheritdoc cref="IDateNanosProperty.NullValue"/>
+		DateTime? IDateNanosProperty.NullValue { get; set; }
```

Comment:
explicit interface member implementations don't need `/// <inheritdoc />` as they already have the doc 😃